js/ui/file_manager/ui.file_manager.files_tree_view.js

```diff
@@ -325,7 +324,7 @@ class FilesTreeViewModel {
             return this._expandAndGetChildrenForGeneralItem(item);
         });
 
-        return whenSome(deferreds)
+        return when.apply(null, deferreds)
```

Comment:
We need to wait all errors (rejects) before selected item loading, because the errors can correct selected item. E.g. if selected folder is not found, then we need select the parent folder.

Reply:
The whenSome method internally call the when.apply(null, deferreds) method. What is the differences?

Reply:
The difference is as follows: `when()` has a fail-fast behavior, while` whenSome()` has a fail-fast behavior.
Actions that are performed after the function `whenSome()` in `_expandAndGetChildrenForRootItem()` method should occur after all the preceding actions are completed, regardless of the status of their completion (success or error). This is necessary in order to keep the selected folder up to date.

Reply:
Ok, I have reverted the code.

Reply:
A more detailed description of the fail-last approach is in[ this article](https://nmaggioni.xyz/2016/10/13/Avoiding-Promise-all-fail-fast-behavior/).